game-core/src/main/java/games/strategy/engine/chat/ChatMessagePanel.java

```diff
@@ -264,7 +264,9 @@ public void addMessageWithSound(final String message, final String from, final b
   }
 
   private void addChatMessage(final String originalMessage, final String from, final boolean thirdperson) {
-    final String message = Ascii.truncate(originalMessage, 200, "...");
+    final String message = from.equals(chat.getServerNode().getName())
+        ? originalMessage
+        : Ascii.truncate(originalMessage, 200, "...");
```

Comment:
This is kind of a hack, but it was the only thing I could come up with without breaking lobby compatibility.

Error logs with a stack trace typically exceed the 200-character limit and so would be truncated by this code.  This solution simply avoids truncation if the message came from the server.

There is precedent for this in the `ChatMessagePanel#addServerMessage()` method.  This method is used to display messages, such as the greeting message from _lobby_server.yaml_, that may be longer than 200 characters.  However, that method is only available on the client, and there's no current mechanism for communicating over the network that a message sent by the server node should not be truncated.

Reply:
This should be fine, it would be nice if this would be documented by a comment here for future reference.

Reply:
06d75fd64